### ISSUE TITLE
Remove publications for which the imported data is no longer in the source

### ISSUE THREAD

**Issue by anaelizabethenriquez** [psu-libraries/researcher-metadata#944]
Depends on #1090 

It happens fairly regularly that Pure makes a mistake about a publication, incorrectly associating it with a Penn State author whose name is similar to one of the publication's authors. Even if the error gets corrected in Pure, it persists in RMD until an admin manually deletes that publication record. (Or just the "authorship" associated with the erroneous author. Note that the publication shouldn't really be deleted unless the erroneous author is the only author at Penn State, but this is virtually always the case.) The same thing can happen with Activity Insight as users sometimes create publication records based on the same kind of incorrect  name matching and author association.

We need an automated process that can iterate over all of the publication records in RMD and decide whether to keep or delete the publication according to the following logic:

**For publications with only Pure import(s):**
  - if the publication is still in Pure (per any associated Pure import record(s))
    - Leave the publication and authorships in RMD. Possibly the update authorship information if we think that we need to account for cases where the publication could have been previously misattributed to a different PSU author with a similar name.
  - if the publication is no longer in Pure (per any associated Pure import record(s))
    - if none of the authors in RMD’s authorship records are currently in Pure
      - Assume that the publication is missing because the author(s) left PSU. Leave the publication and authorships in RMD.
    - if at least one of the authors is still in Pure
       - Assume that the publication is missing because it had been mistakenly attributed to a PSU author and that the mistake has been fixed. Remove the publication from RMD.

**For publications with one or more AI imports and zero or more imports from other sources:**
  - if the publication is still in AI (per any associated AI import record(s))
    - if the status is still "Published" or "In Press" on at least one of the publication records in Activity Insight
      - Leave the publication and authorships in RMD.
    - if the status is not "Published" or "In Press"
      - Remove the publication from RMD
  - if the publication is no longer in AI (per any associated AI import record(s))
    - Assume that the publication is missing because it had been mistakenly attributed to a PSU author and that the mistake has been fixed. Remove the publication from RMD.

For now, this process will be triggered manually as needed. The default behavior for the process should be to perform a "dry run" and simply output the list of publication records that would be deleted from RMD if the process were actually run. Publication records shouldn't actually be deleted from RMD unless a flag or confirmation is provided when the process in invoked.

**Comment by anaelizabethenriquez** [comment]
related to #772 and #147

**Comment by EricDurante** [comment]
I think I'd like to combine #944, #772, and #147 into a single issue since they all seem to be dealing with the same fundamental problem, and there is considerable overlap. That said, I'm not sure that the sum of these issues comprehensively describes the problem or the best solution.

### Framing the problem

The most concrete description of a problem that I could extract from these three issues is from the issue description above and from the discussion around #147:

Pure may occasionally mistake one author (often a non-PSU author) for another PSU author who has a similar name and then incorrectly attribute a publication to that PSU author. That leads to a publication being added to RMD with incorrect authorship information. In every case that we've seen, that publication shouldn’t exist in RMD at all because it doesn’t actually have any PSU authors. Pure may correct this information eventually (by deleting its own record of the publication along with the incorrect authorship information), but RMD's current Pure import processes do not automatically correct the bad information in RMD after Pure fixes the source data.

There are hints of other problems, but they aren't well-described, and we don't seem to be as certain about them. There are a handful of things that I need to clarify to understand the full scope of the issue.

### Open questions

1. We understand one reason (described above) why publications are sometimes removed from Pure. Do we know of any other reasons why publications may be removed from Pure or why authorship information may be changed for a given publication?
2. Likewise, under what circumstance(s) would a publication be removed from Activity Insight in such a way that it should also be removed from RMD? Since Pure and Activity Insight have very different methods of obtaining their data, presumably we wouldn't have the same issue in Activity Insight where non-PSU publications are being added to the dataset due to incorrect author attribution.
3. It has been suggested that it's possible that correctly-attributed publications may sometimes be removed from Pure when they actually shouldn't be. Hence we shouldn't completely delete publications from RMD just because they've been removed from Pure in case they were actually legitimate. However, this seems like it might just be speculation. Do we have any examples of this ever happening in practice? If there are examples, then were the legitimate publications removed from Pure permanently, or were they removed temporarily and added again later?


@anaelizabethenriquez & @nmg110, I have additional thoughts, but let's start with a discussion of these questions.

**Comment by nmg110** [comment]
We could meet with the Pure team to discuss why publications may be removed. I don't know what they do with faculty information if the faculty are terminated or leave. I just know there have been times when a publication has been entered in Pure for a common name, and it wasn't that faculty member's publication. Typically, when this happens, I verify the publication is no longer in Pure or "AI" and will just delete the publication from RMD. This only happens if faculty are using RMD for their department profiles.  

I cannot remember a time when a publication in Pure was removed if it was correct.

In Activity Insight, we disable the account if the faculty leave or are terminated; this way, their information can stay in the system if it is needed for unit-wide reports later. However, if there is no data in their record, meaning they didn't add any publications or presentations, we will delete the user's account to free up licenses in "AI". 

We sometimes have an issue with the wrong publication being entered in "AI" because the staff (proxies) import the publication data from Scopus, PubMed, and/or Crossreff and do not verify the publication. This happens with common names, mostly in the College of Medicine. Once it is deleted in "AI" I don't think it gets removed from RMD.

I hope this helps.

**Comment by anaelizabethenriquez** [comment]
Thank you very much for this, @EricDurante and @nmg110 

1. The wrong-but-similarly named author problem is an error by Pure that gets corrected. I suppose there could be error corrections resulting from other types of errors, but I'm not specifically aware of any. Pure (or more accurately Scopus, which I believe has an identical database of publications) periodically deindexes journals. I'm poking at their documentation to see what happens to publications previously in the database if their journal is deindexed. I wonder if those publications could be deleted. Ideally we would not delete them from RMD, but this is likely a small-percentage case. The issue Nicole raises, that a researcher's publications or authorship might no longer show up in Pure after they leave the university, is important. We would not want those publications to be removed from RMD.
2. For Activity Insight, if the user (either the faculty member or a proxy) deletes a publication in "AI" or regresses the publication status field ("status" in RMD) such that it no longer meets criteria for import, we should delete it in RMD. This could happen if the publication was entered by mistake or if that status field was inaccurately set. Anecdotally, we get a decent number of responses to reminder emails saying "this was never published." Sorry to say I don't have good data on this to assess how many records are likely impacted.
3. IMO this would mostly be a question of the deindexing mentioned above. Possibly moot depending on what that means for the data. Will add documentation here if I can find it.

Nicole wrote: "This only happens if faculty are using RMD for their department profiles." I thought this was more widespread and was due to errors originating in Scopus. Some recent examples in case they are helpful for Nicole:
https://psu.libanswers.com/admin/ticket?qid=12668421 (from Curriculum and Instruction, which I don't think uses RMD for profiles)
https://psu.libanswers.com/admin/ticket?qid=13034421 (College of Medicine)
https://psu.libanswers.com/admin/ticket?qid=11109498 (College of Medicine)

I may need to retract this statement in #772: "One particular thing to consider is publications in RMD that have the status "In Press" or otherwise don't meet the current criteria for import into RMD." We actually are importing "In Press" items, right? What is the current situation with that?

**Comment by EricDurante** [comment]
Thanks, @nmg110 and @anaelizabethenriquez - that's a lot of important information that I didn't have before.

So I see that one of my assumptions was wrong - Activity Insight actually has the same problem with incorrect author attribution that Pure has. That leads me to another question:  have you seen any cases where Pure and AI have made exact same mistake? Or do you think that it's possible that such cases exist? It seems to me that's it's definitely possible.

To answer your question, @anaelizabethenriquez:  yes, we import "In Press" publications from AI:  https://github.com/psu-libraries/researcher-metadata/blob/8458fffc58aef46c64807b1a3cfc9c8981ff8da6/app/importers/activity_insight_importer.rb#L796-L798

**Comment by anaelizabethenriquez** [comment]
Thanks for the answer, Eric!

"So I see that one of my assumptions was wrong - Activity Insight actually has the same problem with incorrect author attribution that Pure has. That leads me to another question: have you seen any cases where Pure and AI have made exact same mistake? Or do you think that it's possible that such cases exist?" => I agree this is likely possible, but defer to @nmg110. I don't have specific examples.

**Comment by nmg110** [comment]
Pure and AI could have the exact same mistake, not sure if it would be for the same publication or not. Anything is possible when humans are involved.

**Comment by EricDurante** [comment]
Quick questions, @anaelizabethenriquez:

What does it mean when a journal is "deindexed" in Scopus? When and why would this happen? Pure, I believe, maintains a list of journals - individual records that have their own metadata about the journal itself. If a journal is "deindexed", does its record get removed?

**Comment by anaelizabethenriquez** [comment]
@EricDurante Here is [Scopus's documentation on "Title re-evaluation"](https://www.elsevier.com/products/scopus/content/content-policy-and-selection#2-title-re-evaluation), which leads to a title (i.e. journal) being removed from the index.

Looking at the spreadsheet attached at the bottom of that documentation section, I see a tab for "Discontinued Titles Nov. 2024", and it includes some date columns with the following text above them:

> FINAL COVERAGE
Disclaimer: Final coverage indicates the last complete volume/issue covered in Scopus.com. After this, no content will be added. No gap fill will be done unless it can be proven that the article was published in one of the issues before the indicated final coverage vol/issue and it was available online before the title was put on hold.

I think this means that Scopus would not remove the journal record you're talking about or the article-level records from prior to the dates in those columns. It just would not include articles going forward. There are some leaps here, but I think this means that no records imported to RMD from Pure would be in danger of deletion or modification due to these rules. So let's say this is moot.

**Comment by EricDurante** [comment]
Ok, given everything that we've discussed so far, I can imagine an automatic process that runs periodically and iterates over each publication record in RMD, re-checks the original data source, and applies logic that goes something like this:

**For publications with only Pure import(s):**
  - if the publication is still in Pure
    - Leave the publication and authorships in RMD. Possibly the update authorship information if we think that we need to account for cases where the publication could have been previously misattributed to a different PSU author with a similar name.
  - if the publication is no longer in Pure (per any associated Pure import record(s))
    - if none of the authors in RMD’s authorship records are currently in Pure
      - Assume that the publication is missing because the author(s) left PSU. Leave the publication and authorships in RMD.
    - if at least one of the authors is still in Pure
       - Assume that the publication is missing because it had been mistakenly attributed to a PSU author and that the mistake has been fixed. Remove the publication from RMD.

**For publications with one or more AI imports and zero or more imports from other sources:**
  - if the publication is still in AI
    - if the status is still "Published" or "In Press"
      - Leave the publication and authorships in RMD.
    - if the status is not "Published" or "In Press"
      - Remove the publication from RMD
  - if the publication is no longer in AI (per any associated AI import record(s))
    - Assume that the publication is missing because it had been mistakenly attributed to a PSU author and that the mistake has been fixed. Remove the publication from RMD.

I think this might be a feasible path for RMD to actually remove bad data and to do so automatically without review or confirmation by an admin. I'd be somewhat hesitant to give RMD admins the burden of determining why publications have gone missing from the original data sources and confirming that they should actually be removed from RMD - mostly just because I don't know how much work that would be. However, I also don't know if the logic above is totally sound or if we could trust anything else that we might come up with until we see how it performs in the real world.

If we do decide that we want to be safe and leave publications/authorships in RMD and queue them up for admin confirmation, then I don't think that using the existing authorship confirmation feature would be the best way to do that. Currently that feature is used for cases where we think that the authorship information is _correct_, but we're not certain. If we also used it here, then we'd be adding in cases where we think that the authorship information is _incorrect_, but we're not certain. We'd then have no way to distinguish between those different cases which we may want to handle differently. So I think it's likely that we'd want to have a different way of flagging and hiding these publications that would keep them in a separate list for admin confirmation - a "confirm to remove" list vs. a "confirm to keep" list. Also, I think that we'd want this flag to be on the publication rather than the authorship records.

Thoughts?

**Comment by anaelizabethenriquez** [comment]
Thanks, @EricDurante ! I agree with everything you wrote about admin confirmation, including that we should not attempt it for this. 

Your logic in the bullet points also looks good. There's one place where it might be a little more complicated. On "if the publication is still in AI", the "(per any associated AI import record(s))" caveat should also apply. And I think we'd want to say "if the status is still "Published" or "In Press" **on at least one import**" leave it in RMD. This would account for the situation where the imports have statuses in different categories.

**Comment by EricDurante** [comment]
> On "if the publication is still in AI", the "(per any associated AI import record(s))" caveat should also apply.

Yes, definitely. Sorry, that was implied in my mind, but better to be explicit.

> I think we'd want to say "if the status is still "Published" or "In Press" on at least one import" leave it in RMD.

Right. Good call.

**Comment by EricDurante** [comment]
Bad news. I've done some investigation, and I don't think that the solution that I proposed above is going to work the way that I was originally picturing.

The ideal way for this all to work would be if the data sources could notify us whenever something is deleted from their databases. I'm thinking of something exactly like the notifications that I just implemented between ScholarSphere and RMD. Unfortunately, as far as I'm aware, Activity Insight has no such capability and neither does Pure.

The next best thing would be for RMD to poll the data sources' APIs on a regular basis to ask if each publication that is in RMD is still present in the corresponding data source. That's essentially the solution that I described earlier in the comments on this issue. Pure's API provides what we would need to make this work - the ability to look up individual publication records by their identifiers. However, as far as I can tell, Activity Insight's API does not provide this ability. Even if it did, Activity Insight has an API rate limit of 10,000 requests per day, and we'd need to make over an order of magnitude more requests than that to check each publication individually. So, it doesn't look to me like the Activity Insight portion of my solution is technically possible.

I've been kicking around other ideas for how we might solve this, and I think I have one that might work, although it's a little contrived. We could essentially cache the current list of publications that are in Activity Insight as we do each import and then check against that cache.

The existing Activity Insight import process in RMD iterates over all of the users in Activity Insight, and for each of those users it iterates over all of the user's publications. As it does this, it updates existing publication records and creates new publication and publication import records for publications that don't already exist. We could add some logic to this part of the import process that captures the current list of all publications each time the importer runs. In terms of additional data that we'd record in RMD, I'm thinking that we'd create a table that represents importer runs - so it would maybe just have a timestamp for when the import process began and another for when it finished. Then we'd create another table that represents individual publication records in Activity Insight. This table would just contain the ID of the publication record in Activity Insight, the current status of the publication (published, in-press, etc.), and a foreign key to the record of the corresponding importer run.

If we captured that data during each import, then we'd be able to compare the publications currently in RMD with the publications in Activity Insight at the time of the last import without making any additional API calls to Activity Insight. So RMD could then have a regularly-scheduled background process that works just like the solution that I described previously except that it checks for each publication in RMD's own record of the most recently completed AI import instead of trying to check directly with AI.

We could do essentially the same thing with the Pure Imports. That might be better than polling since it would avoid making a ton of API calls to Pure.

If this is what we want to do, then we could break this down into at least two separate issues - one for creating the cache of current publication lists and another for the process that checks the cache and deletes bad publication records.

@anaelizabethenriquez How does all of this sound?

**Comment by EricDurante** [comment]
> This sounds like even more work than previously envisioned. Am I right about that?

Yes, it would be some additional work beyond what I proposed previously. Not a huge amount, but a significant amount. Perhaps another day or two of development.

**Comment by anaelizabethenriquez** [comment]
A quick note documenting things we discussed at kick-off today: We are going to proceed with this. We'd like to set this up so that it is run manually by a developer for now (rather than running automatically on a schedule). We'd like to get a report in advance so that @nmg110 and I can review the publications that are going to be deleted, prior to the deletion task actually running.

**Comment by EricDurante** [comment]
I updated the description for this issue and split out #1090 for caching the list of publications in Pure and AI during each import.

**Comment by EricDurante** [comment]
After coding up the logic for handling publications imported from Pure, here is a list of the IDs of all of the publications in RMD that would be deleted as of last week:

`518, 544, 1421, 2936, 3893, 6768, 11069, 11408, 15076, 15537, 20536, 20686, 20978, 21102, 28220, 29943, 30999, 32937, 33345, 34601, 36040, 37870, 37921, 39891, 39971, 40393, 40407, 40451, 42281, 45698, 45920, 47071, 48141, 50521, 51508, 52010, 52172, 52854, 53162, 53228, 53254, 53318, 53508, 53565, 53934, 55772, 57036, 57848, 58736, 59326, 59590, 59788, 59857, 59966, 60046, 61166, 61302, 62342, 63143, 64668, 65137, 67386, 68356, 70154, 71104, 73309, 74241, 75094, 76989, 79535, 79560, 84286, 84326, 86026, 87793, 88569, 89986, 90190, 94464, 95154, 95863, 96637, 98452, 99375, 103294, 103961, 105682, 105965, 106355, 108404, 108840, 111175, 111750, 113881, 115105, 119208, 122117, 122354, 122455, 123248, 123351, 125108, 128108, 128202, 129882, 133098, 220810, 221324, 221513, 221589, 222079, 222554, 223080, 225164, 225853, 226352, 226742, 228110, 228233, 246388, 246451, 246459, 246478, 246490, 246503, 246529, 246585, 246629, 246647, 246698, 246741, 246774, 246791, 246841, 246890, 246945, 247300, 249135, 249463, 250282, 250383, 250428, 250554, 250562, 250649, 250676, 250767, 250789, 250840, 250856, 250890, 250895, 251360, 251387, 251447, 251484, 251550, 251686, 251851, 251871, 251883, 251942, 251954, 251980, 252885, 253323, 253510, 253531, 253558, 253611, 253630, 253680, 253689, 253732, 253797, 253804, 253832, 254040, 254046, 254109, 254110, 254182, 278714, 279459, 285659, 285863, 286153, 287077, 294483, 348710, 348788, 348910, 349230, 349612, 350198, 350323, 350546, 350923, 350971, 351175, 351325, 352308, 352383, 352933, 353063, 353261, 353336, 353709, 353822, 354402, 354539, 354551, 354646, 355475, 355489, 355519, 355551, 355589, 355676, 355963, 356324, 356425, 357551, 357692, 358103, 358278, 358670, 358960, 359205, 359969, 360375, 360750, 360761, 360809, 360874, 361062, 362172, 362815, 362940, 363107, 364371, 364803, 364888, 364942, 365149, 365264, 365520, 365693, 365804, 366155, 366506, 366507, 366614, 367176, 367299, 367403, 367787, 368166, 368390, 369007, 369096, 369202, 370218, 370259, 370318, 370324, 370442, 370503, 370872, 370905, 371076, 371077, 371335, 372222, 372346, 372791, 372821, 373621, 374296, 374437, 374501, 374530, 374933, 375154, 375208, 375965, 376027, 377004, 377546, 377570, 377716, 378708, 378740, 379480, 379620, 380062, 383051, 383444, 383749, 384871, 384927, 385190, 385192, 385194, 385198, 385200, 385202, 385204, 385208, 385212, 385217, 385219, 385223, 385226, 385227, 385228, 385230, 385231, 385387, 386640, 386955, 387154, 387162, 387373, 387627, 387968, 388751, 388871, 389736, 390582, 390819, 390833, 391344, 392576, 392979, 396095, 396159, 396284, 396486, 396540, 396697, 396853, 396922, 397476, 397681, 397736, 397914, 399300, 400043, 400831, 401540, 402402, 402491, 403445, 403523, 403557, 403938, 404063, 404097, 408714, 408738, 408890, 410131, 410224, 410242, 410308, 410782, 411147, 411190, 411210, 411269, 411321, 411551, 411780, 413899, 414189, 414676, 420164, 420181, 420586, 420668, 420773, 421014, 421142, 421164, 421412, 421639, 421644, 421697, 422611, 423746, 424607, 429697, 430824, 431296, 431525, 431979, 433887, 436391, 436411, 437682, 437710, 438962, 440168, 442388, 442524, 444800, 446943, 447102, 447234, 447316, 453328, 453471, 453495, 453520, 453554, 454266, 454326, 454331, 454334, 454359, 454401, 454421, 454446, 454493, 454550, 454578, 454619, 454636, 454643, 454675, 454680, 454722, 454755, 454757, 454762, 454774, 455895, 455934, 455945, 456048, 456206, 456229, 456300, 456301, 456306, 456307, 456313, 456333, 456345, 456360, 456395, 456425, 456428, 456443, 456463, 456512, 456524, 456548, 456600, 456659, 456671, 456680, 456682, 456688, 456694, 456718, 456719, 456726, 456729, 456737, 456767, 456787, 456794, 456839, 456865, 456879, 456880, 456898, 456909, 456937, 457015, 457070, 457076, 457080, 457145, 457153, 457181, 457183, 457226, 457232, 457235, 457275, 457295, 457307, 457309, 457328, 457367, 457379, 457795, 457902, 458694, 458728, 508626, 508634, 508640, 508658, 508659, 508665, 508669, 508670, 508672, 508673, 508674, 508679, 508680, 508683, 508685, 508702, 508713, 508720, 508724, 508731, 508737, 508744, 508745, 508750, 508751, 508752, 508753, 508854, 508962, 508964, 509208, 509292, 510285, 510290, 510389, 510476, 510683, 512456, 513574, 515584, 519969, 520377, 520408, 520687, 520746, 520891, 520971, 521166, 524378, 525835, 529900, 530170, 534915, 534917, 534934, 534935, 534939, 534944, 534948, 534956, 534962, 534966, 534975, 534978, 534979, 534983, 543481, 543527, 546117, 546301, 546522, 547111, 547218, 547378, 549323, 553066, 553133, 553160, 553634, 553730, 553769, 553774, 553897, 554110, 554314, 554374, 554476, 554610, 555213, 555297, 555934, 555935, 555936, 555989, 557381, 560827, 561052, 561061, 562763, 568239, 574705, 579989, 582649, 583401, 583494, 583995, 584042, 584864, 585134, 586009, 586866, 586875, 587656, 588737, 588785, 588796, 588843, 595236`

Response from @anaelizabethenriquez:

> Want to share a couple unexpected things in case they mean something to you.
>
> Pub 518 (or a pub with the same metadata) appears to be in the RMD user's Pure profile: 
> https://pure.psu.edu/en/publications/acute-fatty-liver-of-pregnancy-and-longchain-3hydroxyacylcoenzyme
>
>The same is true for 1421: https://pure.psu.edu/en/publications/a-marker-of-early-amacrine-cell-development-in-rat-retina-2

**Comment by EricDurante** [comment]
@anaelizabethenriquez 

That didn’t immediately mean anything to me, but I’ve done some investigation. Looking at the publication 518 in RMD:
- This publication has one associated import record, and that record is from Pure.
- The record in Pure that is referenced by the import record in RMD indeed no longer exists.
- This publication has a DOI recorded in RMD that does not resolve.
- This publication is associated with a single author, dxh402

There is another publication record in RMD with ID 119093. This publication:
- has a title that appears identical to the title of publication 518 when it’s printed out, but actually differs in that it uses multi-byte hyphen characters vs. the single-byte hyphen characters used in 518's title
- has a different DOI than publication 518, and this DOI does apparently resolve to the publication
- is also associated with the same single author (dxh402) as pub 518
- also has a single associated publication import record that is from Pure and which references a record that is still currently in Pure with metadata matching publication 119093 in RMD

So, it looks like RMD’s publication 119093 is the one that you’re seeing [here](https://pure.psu.edu/en/publications/acute-fatty-liver-of-pregnancy-and-longchain-3hydroxyacylcoenzyme) and that publication 518, which had similar but possibly incorrect metadata, is missing from Pure and should be deleted from RMD. However, this isn’t the scenario that we were originally solving for. Instead of publication 518 having been misattributed to a Penn State author, it looks like it maybe had some bad metadata, or metadata that changed at some point, and instead of correcting their original record, Pure deleted the original record and created a new one.

It looks like something similar is going on with publication 1421, in which case publication 24813 in RMD is referencing the new record in Pure.

**Comment by anaelizabethenriquez** [comment]
595236 appears to be an example of the problem we were trying to solve. The article has an author with a similar name to a PSU researcher, but based on the PSU researcher's CV and the affiliation listed in the published article, it seems to refer to a different person.

588843 and 374933 appear to follow the pattern in Eric's comment above.

I also reviewed 250428. I'm having difficulty confirming it as belonging in either of the categories above. The author was not affiliated with Penn State at the time the article was published. It seems plausible that it's in the "Eric's comment" pattern, but I'm not certain.

So far everything I've looked at has been OK to remove. At the pace I'm going, I don't have time to review even 10% of the 621. I'd lean toward moving forward. I'm wondering @EricDurante if you have any ideas about how I might review these more systematically/faster, or if you think we should move ahead.

**Comment by EricDurante** [comment]
Thanks, @anaelizabethenriquez. I was just looking for some spot-checking to make sure that things are generally working as we intended and that we didn't get something totally wrong. It looks like that's the case, so I do think that we can move ahead.

**Comment by EricDurante** [comment]
Here is a sample of RMD IDs from the list of publications imported from Activity Insight that would be deleted according to the logic for that part of this process. There are 5,625 publications currently in that list.

Here's the beginning of the list:
`240, 591, 1053, 1073, 1401, 1415, 1489, 1501, 1873, 2102, 2494, 2551, 2972, 3124, 3187, 3202, 3409, 4075, 5238, 5377`

A couple of samples from the middle:
`234717, 234808, 235331, 235332, 235408, 235436, 235633, 235973, 236060, 236069, 236075, 236079, 236791, 236793, 236803, 236813, 236821, 237184, 237204, 237210, 237216`

`446225, 447015, 448303, 448629, 448651, 448664, 448666, 448667, 448668, 448669, 448672, 448674, 449244, 449245, 449258, 449260, 449263, 449264, 449408, 449409, 449410`

And the end of the list:
`601802, 601803, 601804, 601805, 601806, 601807, 601808, 601809, 601810, 601811, 601812, 601813, 601814, 601815, 601816, 601817, 602347, 602598, 603157, 603314, 603319`

**Comment by nmg110** [comment]
I say, go for it. The ones I looked at look like they have been removed from "AI" for one reason or another. :-)

**Comment by EricDurante** [comment]
Right away I'm noticing that at least some of these are publications that have merged with imports from Pure, and that the publications still exist in Pure (I just checked the first two, 240 and 591). Here's the publication in Pure that's linked to 591:  https://pure.psu.edu/en/publications/hydrothermal-gasification-of-nannochloropsis-sp-with-ruc

@nmg110 or @anaelizabethenriquez Any idea why these publications were would have been removed from Activity Insight? It doesn't seem to me like these records should be deleted from RMD. Do you think that we should not be deleting publications that are now missing from AI if those publications are also linked to something that's currently in Pure?

**Comment by EricDurante** [comment]
447015 is a case where the publication is still in Activity Insight with status "Submitted", but it's also still in Pure with status "Published". Is the status in Activity Insight simply not up-to-date in this case? This seems like another situation where we might not want to delete the publication from RMD.

**Comment by EricDurante** [comment]
Just a couple of notes:

448303 and 448629 look like a legitimate cases where there is only an Activity Insight import, and the statuses regressed from "In Press" or "Published" to "Submitted".

448651 along with some others that I've checked also look like legitimate cases where there is only an Activity Insight import and the record was removed from Activity Insight altogether.

**Comment by anaelizabethenriquez** [comment]
> Do you think that we should not be deleting publications that are now missing from AI if those publications are also linked to something that's currently in Pure?
Yes. If the publication is also linked to something that is currently in Pure, we should not delete the publication record. (I realize we did not account for this in the issue description.)

> 447015 is a case where the publication is still in Activity Insight with status "Submitted", but it's also still in Pure with status "Published". Is the status in Activity Insight simply not up-to-date in this case? This seems like another situation where we might not want to delete the publication from RMD.
Yes, I suspect Activity Insight just isn't up to date. I agree we wouldn't want to delete this publication from RMD. Since it has a Pure record associated with it, I think it's covered by the case above. If it was not merged with a Pure record in RMD, I would want to delete it because its status in Activity Insight is not "In Press" or "Published," so it doesn't qualify for import under our current rules. Since it's truly published, we actually would want it in RMD, but since AI isn't telling us that, we have no way to know.

However, this has made me realize that we should add additional conditions on whether a publication can be deleted. (Sorry! I know it is late in the game to be saying this.) Regardless of everything else in this issue, we should not delete publications that:

- Have an associated "Activity insight OA file" record in RMD, and/or
- Have an open access location with a source of "User" or "ScholarSphere", and/or
- Have an associated waiver
- (I hope that's a complete list.)

The reasoning here is that if one of the above applies, we need to keep a record of it. The existence of one of these things also suggests the publication info is actually appropriate to be in RMD. If it is relatively easy to generate a list of these, I would love to see it, but no worries if not.

**Comment by EricDurante** [comment]
> However, this has made me realize that we should add additional conditions on whether a publication can be deleted. (Sorry! I know it is late in the game to be saying this.)

No, I wouldn't say that it's late in the game. This is so complex that there was no way that we were going to think of everything or anticipate all of the possible issues up front. We're just working through the exploration and iteration that was going to be required to get this right.

So, it's clear that we also need to amend our original logic to account for unexpected situations involving merged publications. Here's what I'm thinking that we need to change it to (in plain-English pseudocode that is as concise as I can make it):

```
for all publications that have at least one pure import or at least one AI import and no imports from other sources, delete the publication only if it:

(has no Pure import OR (has Pure import(s), none of which are currently in Pure AND has at least one author who is in Pure)) AND (has no AI import OR has AI import(s), none of which are currently in AI with a status of “In Press” or “Published”)
```
@anaelizabethenriquez Does that seem more accurate if we also apply the additional conditions that you listed above to filter out those publications from the set that we're initially considering?

The only potential issue that I see with this logic is that a current, published record in either place (Pure or Activity Insight) would keep the record from being deleted in RMD. So for the misattribution problem, this would mean that if the same misattribution was made for the same publication in both Pure and Activity Insight, then it would need to have been corrected in both places for the record to be removed from RMD. However, that actually seems pretty reasonable to me especially considering that the double-misattribution situation seems unlikely to occur often, and even if/when it does, it should end up getting fixed in both places.


> If it is relatively easy to generate a list of these, I would love to see it, but no worries if not.

Do you mean one list of all of the publications for which any one of those three conditions is true?

**Comment by anaelizabethenriquez** [comment]
@EricDurante Thanks. That pseudocode, plus the additional conditions I listed, seems good to me. I am OK with how this would work in cases of double-misattribution.

I meant a list of all of the publications _that would otherwise be deleted_ for which any one of those three conditions is true. A list of publications that would otherwise be deleted for which a specific one of those conditions is true would also be helpful.

**Comment by anaelizabethenriquez** [comment]
Great! Definitely no need for three separate lists.

**Comment by EricDurante** [comment]
@anaelizabethenriquez & @nmg110, I've discovered another situation that's unexpected (for me at least). There are apparently cases where a record in Activity Insight originally had the status "Published" and then the status was removed completely (it's now NULL). We're treating these publications in the same way that we are if the status changed to any other value that's not "Published" or "In Press" - that is, we'd be deleting them. Do you think that's what we should be doing with these?

**Comment by anaelizabethenriquez** [comment]
Sounds right to me, though I defer to @nmg110 on this. I just tested in my own Activity Insight account, and I'm able to delete the status entirely and save the record (retaining the rest of the metadata). I imagine someone might do that if they turned away from a project or things fell apart (even if they had optimistically or accidentally marked it as In Press or Published prior to that). I think it's a clear sign that it doesn't belong in RMD.

**Comment by nmg110** [comment]
Yes, you can delete them. Once they are corrected in AI, they will be imported back in, right?

**Comment by EricDurante** [comment]
@anaelizabethenriquez, those additional filters removed 104 publications from the list to be deleted. Those publications are:

`191992, 191993, 278250, 290006, 290007, 290009, 292128, 292129, 292131, 292651, 293246, 293247, 333498, 333499, 333500, 333501, 333502, 333503, 333504, 333505, 348788, 350923, 370318, 403557, 413051, 413052, 413053, 413054, 413056, 413057, 413058, 413059, 413060, 413446, 415864, 415865, 416518, 416689, 417706, 418739, 419675, 422927, 422987, 438962, 440168, 444004, 444117, 447102, 450684, 453520, 453774, 455895, 455945, 456206, 456333, 456671, 456737, 456865, 456909, 456937, 457015, 457076, 457080, 457181, 457309, 458728, 485176, 505820, 505825, 505826, 505827, 505835, 505836, 505844, 505845, 505850, 505851, 505852, 505856, 505857, 505862, 505863, 505864, 505866, 505867, 511547, 527807, 529107, 529900, 534939, 534944, 534956, 534975, 534978, 536766, 537214, 537215, 537678, 555989, 570052, 581140, 581141, 592960, 598839`

**Comment by anaelizabethenriquez** [comment]
Thanks, @EricDurante. This looks good to me. I've spot-checked those and for every one I've checked it makes sense not to remove it. I will also make time soon to go through all of them and manually curate as appropriate.

**Comment by ajkiessl** [comment]
@anaelizabethenriquez I ran the deletion script in QA.  It failed while trying to delete a publication that had a scholarsphere deposit (the association prevented it from deleting).  The publication ID that caused the failure is 409694 .  I'm assuming we don't want to delete publications if they have a scholarsphere deposit, so I'm going to adjust the code to filter them out.  Just wanted to leave a comment here in case something about this doesn't sound right to you.

**Comment by anaelizabethenriquez** [comment]
@ajkiessl Thanks. I'm not certain why the ScholarSphere OA location is missing on this publication. There is a mismatch (due to character formatting) between the title in RMD and the title in ScholarSphere. Also, the DOI is missing in RMD. I will fix both those issues. Nonetheless, given the presence of a Scholarsphere work deposit, this file was deposited using RMD, so I would have expected an OA location with source of ScholarSphere. I'll fix that now too.

> Maybe we just want to fix the data, rather than filtering out things with scholarsphere deposits.
If it's easy for you to generate a list of any other pubs that have this problem, I'm happy to fix them manually (within reason -- I don't anticipate there would be many). That would certainly be preferable. But if timing-wise it's better just to filter these out, that's fine too.

**Comment by anaelizabethenriquez** [comment]
Oops, I need to take some of that back, @ajkiessl. Looks like the associated Scholarsphere deposit is actually for a different article.

**Comment by anaelizabethenriquez** [comment]
OK, here's the scoop. Looks like the file that got deposited to ScholarSphere via RMD has a different title than the RMD record. Rather than withdraw it, the ScholarSphere team corrected the metadata in ScholarSphere to describe the file that was actually deposited. And then I would have (at their direction) deleted the OA location in RMD pointing to ScholarSphere.

I suspect the ScholarSphere deposit is the "same" publication but has the correct final title. However, without knowing more, I wouldn't want normally want to treat this publication as a duplicate of 456477 (which has the title and DOI that match the ScholarSphere deposit). There's just not a good way for me to be sure that 409694 refers to the same publication (unless I contact the author, but even then they might not remember).

409694 is exactly the kind of thing we want to delete. Even the Scholarsphere work deposit associated with it is not a reason to keep it, because that work deposit actually belongs with a different publication.

If it is possible for me to manually review these so that we can actually get them deleted, I think that would be ideal. I could also just put an issue for that in the backlog, with the idea that we'd keep publications for now if they have a Scholarsphere work deposit, but we might manually remove them and update that logic later.

**Comment by ajkiessl** [comment]
@anaelizabethenriquez Ah okay, thanks for digging into this more. 

Yes, it makes sense to report these to you so you can confirm they need to be deleted, then I can delete them from the back-end.  Is it okay if I delete this one now?  The Scholarsphere deposit and the publication?

**Comment by anaelizabethenriquez** [comment]
@ajkiessl Sounds good. And yes, please do delete this publication and the associated Scholarsphere deposit. Thanks!
